src/s3-copy.py

```diff
@@ -1,10 +1,12 @@
 #!/usr/bin/env python3
 
-from datetime import datetime, timedelta
+from datetime import datetime
 import utils.time_utils as time_utils
 import utils.file_utils as file_utils
 import yaml
 import sys
+import boto3
+import botocore
 
 #----- check the python version because this assumes we use ordered dictionaries
 assert sys.version_info >= (3, 6), "Requires python > 3.6"
```

Comment:
I suggest using the `collections` package for ordered dictionaries. There is a method explicitly for this.

Reply:
i am curious. why? 3.6 has been released 6 years ago and i have a bumper preventing me from using versions below 3.6.

Reply:
Either `pip` or `conda` should allow you to build Python libraries that are consistent with your Python build. That will allow you to avoid having to change this if/when you decide to use a more recent Python build. It's not a deal breaker but it is good practice.

Reply:
ah. right. i can require 3.6 or better in the setup.py file. i get that now. Still thinking about this as a script not a package.

Reply:
Sure. You can also keep your Python build and libraries up-to-date. I understand this is a fail safe. It's just something I don't prefer to do myself. User preferences I suppose.